src/foam/nanos/ruler/CompositeRuleAction.js

```diff
@@ -0,0 +1,31 @@
+foam.CLASS({
+  package: 'foam.nanos.ruler',
+  name: 'CompositeRuleAction',
+
+  documentation: 'runs multiple rule actions from one action',
+
+  implements: ['foam.nanos.ruler.RuleAction'],
+
+  javaImports: [
+    'foam.nanos.ruler.RuleAction',
+  ],
+
+  properties: [
+    {
+      class: 'FObjectArray',
+      of: 'foam.nanos.ruler.RuleAction',
+      name: 'ruleActions'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+        for( RuleAction action : getRuleActions() ) {
```

Comment:
Shouldn't this be wrapped in something like the following?

```Java
agency.submit(x, new ContextAgent() {
  @Override
  public void execute(X x) {
    // Your code here
  }
}, "Executing multiple rule actions.");
```

Reply:
Wouldn't each rule action use its own context agent if it requires one?

Reply:
I guess agency.submit would be in the child action itself.

Reply:
👍